### ISSUE TITLE
save wallet file

### ISSUE THREAD

**Issue by TechyGuy17** [kryptokrona/hugin-api#316]
Currently the wallet file never gets saved, so XKR sent to the api is lost on reboot

**Comment by Coffeboi** [comment]
Coffe is on the case

**Comment by f-r00t** [comment]
I believe it's an issue with the encryptWalletToString(). You should use toJSONString() instead.

See here: https://github.com/kryptokrona/hugin-api/blob/126bede038d5eb95cd00fe218764383cdc2610b4/utils/wallet.js#L90

**Comment by n9lsjr** [comment]
I do not think that is the issue, the problem seems to be that the wallet only saves one time (on startup)

I think you need a interval like this to save the wallet more often:
https://github.com/kryptokrona/yggdrasil-wallet/blob/149c335efd6e69ad2c8c307cbfa0fe80677cac7d/src/backend/electron.cjs#L306

Otherwise it will not remember anything that happens during the wallet is online and will thus reset on next startup

**Comment by f-r00t** [comment]
It saves every 60s.

First the saveWallet() function is called on startup:
https://github.com/kryptokrona/hugin-api/blob/main/server.js#L159

And as you can see here, the function then calls itself every 90s:
https://github.com/kryptokrona/hugin-api/blob/126bede038d5eb95cd00fe218764383cdc2610b4/utils/wallet.js#L102

So either that loop crashes, potentially because it gets "too recursive" i.e. is several thousand function calls in eventually, or there is something else wrong.

A better method would probaby be to have an interval that is started in server.js, something like:
`      setInterval(function() {
        saveWallet();
      }, 60000);
      `

**Comment by n9lsjr** [comment]
That is true lol, but in this recursive loop the variable walletExists is set to true after first save, then it will probably not save it again.

https://github.com/kryptokrona/hugin-api/blob/126bede038d5eb95cd00fe218764383cdc2610b4/utils/wallet.js#L97-L99

**Comment by f-r00t** [comment]
Yeah that makes sense. After doing some research on sequalize, it seems like this is the general way of updating an entry:
`// Change everyone without a last name to "Doe"
await User.update({ lastName: "Doe" }, {
  where: {
    lastName: null
  }
});`
(https://sequelize.org/docs/v6/core-concepts/model-querying-basics/)

That probably would translate to something like this in our codebase:
`await sequelize.transaction(async (t) => {
    return models.Wallet.update(
        { encrypted_wallet: newEncryptedStr }, 
        { where: {} } // Empty where clause updates all records, i.e. the only wallet record we have
    );
});`
